.github/workflows/integration_test.yaml

```diff
@@ -0,0 +1,85 @@
+name: Integration tests
+
+on:
+  workflow_call:
+    outputs:
+      images:
+        description: Pushed docker images
+        value: ${{ jobs.get-images.outputs.images }}
+
+env:
+  REGISTRY: ghcr.io
+  OWNER: ${{ github.repository_owner }}
+
+jobs:
+  get-images:
+    name: Get images
+    runs-on: ubuntu-20.04
+    outputs:
+      images: ${{ env.DOCKER_IMAGES }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Get dockerfiles
+        id: set-images
+        run: echo "DOCKER_IMAGES=$(ls *.Dockerfile 2> /dev/null | sed s/\.Dockerfile// |  jq -Rsc '. / "\n" - [""]')" >> $GITHUB_ENV
+  build-images:
+    name: Build image
+    runs-on: ubuntu-20.04
+    needs: get-images
+    if: ${{ needs.get-images.outputs.images != '[]' }}
+    strategy:
+      matrix:
+        image: ${{ fromJSON(needs.get-images.outputs.images) }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and push image
+        id: build
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          push: true
+          tags: ${{ env.REGISTRY }}/${{ env.OWNER }}/${{ matrix.image }}:${{ github.run_id }}
+          file: ${{ matrix.image }}.Dockerfile
+
+  integration-test-microk8s:
```

Comment:
Is this limited to microk8s tests?

Reply:
Tests are run in microk8s within that step

Reply:
I guess the question was, for the Jenkins charm (and other machine charms), the integration tests would run on lxd. I wonder if it is possible to support both?

Reply:
Not for now. I've made some research and the most feasible idea for now is to have two different reusable workflows. I'll work into that when the time to migrate a machine charm comes.